### PR TITLE
chore: migrate nodes to live in baremetal project

### PR DESCRIPTION
**Pull request by cardoe**
Move the project that our servers live in and get controlled in do a new domain called 'infra' with a project called 'baremetal'

### CODE REVIEW

File: components/keystone/aio-values.yaml [hunk 1]
```diff
@@ -10,17 +10,17 @@ bootstrap:
...
-    # allow ironic user to see servers in undercloud project
-    openstack role add --project undercloud --user ironic --user-domain service  member
+    # create 'infra' domain
+    openstack domain create --or-show infra
+    # create 'baremetal' project for our ironic nodes to live in
+    openstack project create --or-show --domain infra baremetal
+    # create 'argoworkflow' user for automation
+    openstack user create --or-show --domain infra --password demo argoworkflow
+    # give 'argoworkflow' 'admin' over the 'baremetal' project
+    openstack role add --user-domain infra --project-domain infra --user argoworkflow --project baremetal admin
```
Comment (nicholaskuechler) [inline components/keystone/aio-values.yaml:20]: I think we still need member role for horizon ui access

File: components/keystone/aio-values.yaml [hunk 2]
```diff
@@ -10,17 +10,17 @@ bootstrap:
...
+
+    # this is too early because ironic won't exist
     openstack role add --project service --user ironic --user-domain service service
-    # add 'demo' user to have 'member' role, needed for horizon dashboard use
-    openstack role add --user demo --project undercloud member
```
Comment (nicholaskuechler) [inline components/keystone/aio-values.yaml:23]: I think we still need member role for horizon ui access

File: components/keystone/aio-values.yaml (hunk 1 above)
Comment (cardoe) [reply components/keystone/aio-values.yaml:20]: Will the `argoworkflow` user be logging into the horizon UI? People should be logging in as themselves and not the service account.

File: components/keystone/aio-values.yaml (hunk 2 above)
Comment (cardoe) [reply components/keystone/aio-values.yaml:23]: Okay here.

Comment (cardoe) [reply components/keystone/aio-values.yaml:23]: I've made `demo` be part of `ucuser`.

**Review by skrobul — APPROVED** [review]
let's see what breaks

**Pull request merged**